File: caesar_cipher.c
// Caesar Cipher Code

#include <stdio.h>
#include <string.h>

void encrypt(char *plainText, int key)
{
    for (int i = 0; i < strlen(plainText); i++)
    {
        if (plainText[i] >= 65 && plainText[i] <= 91)
            plainText[i] = ((plainText[i] - 65 + key) % 26) + 65;

        else if (plainText[i] >= 97 && plainText[i] <= 123)
            plainText[i] = ((plainText[i] - 97 + key) % 26) + 97;
    }
}

void decrypt(char *cipherText, int key)
{
    for (int i = 0; i < strlen(cipherText); i++)
    {
        if (cipherText[i] >= 65 && cipherText[i] <= 91)
            cipherText[i] = ((cipherText[i] - 65 - key + 26) % 26) + 65;

        else if (cipherText[i] >= 97 && cipherText[i] <= 123)
            cipherText[i] = ((cipherText[i] - 97 - key + 26) % 26) + 97;
    }
}

int main()
{
    char text[] = "tanmayvaij";
    int key = 5;

    encrypt(text, key);

    printf("encryted text: %s \n", text);

    decrypt(text, key);

    printf("decrypted text: %s \n", text);

    return 0;
}

File: caesar_cipher.py
def encrypt(plain_text, shift):

    cipher_text = ""

    for char in plain_text:
        cipher_text += chr((ord(char) + shift - 65) % 26 + 65)

    return cipher_text


def decrypt(cipher_text, shift):

    plain_text = ""

    for char in cipher_text:
        plain_text += chr((ord(char) - shift - 65) % 26 + 65)

    return plain_text


plain_text = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
shift = 23

cipher_text = encrypt(plain_text, shift)
print(cipher_text)

decrypted_cipher_text = decrypt(cipher_text, shift)
print(decrypted_cipher_text)
